ts/features/itwallet/identification/screens/spid/ItwSpidIdpLoginScreen.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { memo, useCallback, useMemo } from "react";
+import { Linking, StyleSheet, View } from "react-native";
+import { WebView, WebViewNavigation } from "react-native-webview";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import {
+  selectAuthUrlOption,
+  selectIsLoading
+} from "../../../machine/eid/selectors";
+import { ItwEidIssuanceMachineContext } from "../../../machine/provider";
+import LoadingScreenContent from "../../../../../components/screens/LoadingScreenContent";
+import I18n from "../../../../../i18n";
+import { originSchemasWhiteList } from "../../../../../screens/authentication/originSchemasWhiteList";
+import { itWalletIssuanceRedirectUri } from "../../../../../config";
+import { getIntentFallbackUrl } from "../../../../../utils/login";
+import {
+  HeaderSecondLevelHookProps,
+  useHeaderSecondLevel
+} from "../../../../../hooks/useHeaderSecondLevel";
+
+const styles = StyleSheet.create({
+  webViewWrapper: { flex: 1 }
+});
+
+const LoadingSpinner = (
+  <LoadingScreenContent contentTitle={I18n.t("global.genericWaiting")} />
+);
+
+/**
+ * This component renders a WebView that loads the URL obtained from the startAuthFlow.
+ * It handles the navigation state changes to detect when the authentication is completed
+ * and sends the redirectAuthUrl back to the state machine.
+ */
+const ItwSpidIdpLoginScreen = () => {
+  const isMachineLoading =
+    ItwEidIssuanceMachineContext.useSelector(selectIsLoading);
+  const spidAuthUrl =
+    ItwEidIssuanceMachineContext.useSelector(selectAuthUrlOption);
+  const machineRef = ItwEidIssuanceMachineContext.useActorRef();
+
+  const onError = useCallback(() => {
+    machineRef.send({ type: "error", scope: "spid-login" });
+  }, [machineRef]);
+
+  // To ensure the server recognizes the client as a valid mobile device, we use a custom user agent header.
+  const defaultUserAgent =
```

Comment:
Can we move `defaultUserAgent` outside the component, since it's a constant value?